app.js

```diff
@@ -0,0 +1,113 @@
+const appRoot = require('app-root-path');
+const config = require('config');
+const express = require('express');
+const fs = require('fs');
+const yaml = require('js-yaml');
+const git = require('simple-git/promise');
+const https = require('https');
+const moment = require('moment');
+
+const MAX_PAGE_SIZE = 10000;
+
+const db = appRoot.require('/db/db');
+const { badRequest, notFound, errorHandler } = appRoot.require('/errors/errors');
+const { authentication } = appRoot.require('/middlewares/authentication');
+const { logger } = appRoot.require('/middlewares/logger');
+const api = appRoot.require('/package.json').name;
+const {
+  port,
+  adminPort,
+  keyPath,
+  certPath,
+  secureProtocol,
+} = config.get('server');
+const { basePath } = yaml.safeLoad(fs.readFileSync(`${appRoot}/swagger.yaml`, 'utf8'));
+
+/**
+ * @summary Initialize Express applications and routers
+ */
+const app = express();
+const appRouter = express.Router();
+const adminApp = express();
+const adminAppRouter = express.Router();
+
+/**
+ * @summary Middlewares
+ */
+if (logger) app.use(logger);
+app.use(`/api${basePath}`, appRouter);
+appRouter.use(authentication);
+
+adminApp.use(`/api${basePath}`, adminAppRouter);
+adminAppRouter.use(authentication);
+adminAppRouter.use('/healthcheck', require('express-healthcheck')());
+
+/**
+ * @summary Get application information
+ */
+adminAppRouter.get('/', async (req, res) => {
+  try {
+    const commit = await git().revparse(['--short', 'HEAD']);
+    const now = moment();
+    const info = {
+      meta: {
+        name: 'express-api-skeleton',
+        time: now.format('YYYY-MM-DD HH:mm:ssZZ'),
+        unixTime: now.unix(),
+        commit: commit.trim(),
+        documentation: 'swagger.yaml',
+      },
+    };
+    res.send(info);
+  } catch (err) {
+    errorHandler(res, err);
+  }
+});
+
+/**
+ * @summary Get APIs
+ */
+appRouter.get(`/${api}`, async (req, res) => {
+  try {
+    const { page } = req.query;
+    if (page.size > MAX_PAGE_SIZE) {
```

Comment:
I tested this endpoint with no parameters and the api returned a 500:
```
TypeError: Cannot read property 'size' of undefined
    at appRouter.get (/Users/ianbrown/github/express-api-skeleton/app.js:73:14)
    at Layer.handle [as handle_request] (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/layer.js:95:5)
    at /Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/index.js:275:10)
    at authMiddleware (/Users/ianbrown/github/express-api-skeleton/node_modules/express-basic-auth/index.js:49:16)
    at Layer.handle [as handle_request] (/Users/ianbrown/github/express-api-skeleton/node_modules/express/lib/router/layer.js:95:5)
info: GET /api/v1/express-api-skeleton 500 1ms {"meta":{"res":{"statusCode":500},"req":{"url":"/api/v1/express-api-skeleton","headers":{"cache-control":"no-cache","postman-token":"1218bb70-9994-4c7d-862d-6c4347307792","authorization":"Basic dXNlcm5hbWU6cGFzc3dvcmQ=","user-agent":"PostmanRuntime/7.3.0","accept":"*/*","host":"localhost:8080","accept-encoding":"gzip, deflate","connection":"keep-alive"},"method":"GET","httpVersion":"1.1","originalUrl":"/api/v1/express-api-skeleton","query":{}},"responseTime":1},"timestamp":"2018-09-28T18:14:52.629Z"}
```
Looks like there should be a check that `page` exists

Reply:
Good catch. I've implemented the null-safe check for this. However, it seems like we can take advantages of babel to use the similar syntax as we do in Groovy, e.g. `page?.size`, but I'll need to look into it more. I'll push another commit when I figure it out.